### DISCUSSION TITLE
Server doesn't return any data.

### DISCUSSION THREAD

**Discussion by YLagh01** [maplibre/martin#1287] [Q&A] [answered]
Hi everyone.

My martin server keeps returning 204 HTTP code when I try to test it with QGIS, meaning that there is no data sent from the server. I used the "`mbtiles validate *path*`" command to verify my data integrity and it's OK (I downloaded it from Geofabrik and converted it to MBTiles format with tilemaker). 

I also tried to download the data using `wget http://0.0.0.0:3000/Luxembourg/{z}/{x}/{y}` without success. (Screen of this below)

But when I use MBTiles files containing raster tiles (png), it works perfectly fine and returns png data, which is extremely confusing as I thought that Martin was able to serve only vector data. 

I would really appreciate someone explaining:
-Why is Martin able to return PNG data
-Why is Martin returning 204 code and not sending data when I use it with vector MBTiles files.

Thank you!
![Screenshot from 2024-04-07 19-35-57](https://github.com/maplibre/martin/assets/129891145/b527690e-5772-4cf4-a0b0-013aa74c2426)

**Comment by sharkAndshark** [comment, marked as answer]
Hi @Zit-Zitoun Have a good day!

## Why is Martin able to return PNG data

`Martin` just return the tile data stored in `MBTIles`, and `MBTiles` could store PNG data based on the [spec](https://github.com/mapbox/mbtiles-spec/blob/master/1.3/spec.md#content-1): `The tile_data column MUST contain the raw binary image or vector tile data for the associated tile as a blob`

## Why is Martin returning 204 code and not sending data when I use it with vector MBTiles files.

* Maybe there is no tile in `11/48/54`? 
* What's your tile-grid?
* Have you view it on QGIS?

And would you mind share this file? I'd love to debug it.  

PS. You could inspect the MBTiles file with tools like [DBeaver](https://dbeaver.io/), `Navicat`, etc. They are sqlite file generally.

**Reply by YLagh01** [reply]
Hi @sharkAndshark 
Thank you very much for your quick answer!

I figured out that I was not using QGIS properly, which made me think that the server wasn't functionnal ( I spent a whole day trying to "fix" my server 😅). The database was also fine. 

Sorry for the inconvenience about such a silly mistake and thank you again!

PS: I couldn't share the file because it was too big! Even with compression.

**Comment by radi2015** [comment]
i also encounter same 204 question
requests.get("http://0.0.0.0:3000/tms/10/857/418")
<img width="1234" height="134" alt="image" src="https://github.com/user-attachments/assets/a02d5ea5-6a46-4fdf-9ce2-6d2975d6eabf" />

data is in my mbtiles

<img width="1338" height="158" alt="image" src="https://github.com/user-attachments/assets/08628609-59b8-4999-a149-6257a4a37904" />

**Reply by CommanderStorm** [reply]
Ah, A classic.

Sorry to say, but you are looking at the wrong part of the database.

> `MBTiles` uses a TMS (Tile Map Service) scheme for its tile coordinates (inverted along the Y axis).

=> you need to look at `tile_row=605`

You can invert the value via:

```rust
fn invert_y_value(zoom: u8, y: u32) -> u32 {
    (1u32 << zoom) - 1 - y
}
```

**Comment by radi2015** [comment]
<img width="2120" height="308" alt="image" src="https://github.com/user-attachments/assets/0ac5a4db-6290-4a99-8524-7e15424d1232" />
 terminal is so quiet，no any response

**Reply by CommanderStorm** [reply]
Please refer to https://maplibre.org/martin/troubleshooting.html on how to increase the log level.
